azurerm/helpers/azure/key_vault.go

```diff
@@ -0,0 +1,121 @@
+package azure
```

Comment:
can we also apply these changes to the Key Vault Access Policy resource and the matching Data Sources too?

Reply:
Key Vault Access Policy doesn't use a uri, but the vault name. Would we just be changing it to be consistent and deprecating the `name` property?

Reply:
yep - this bug would still be possible with the Access Policy resource otherwise?

Reply:
I don't believe so, the api it is using is the same one as we use to now check to see if the keyvault exists.

Reply:
this is less the API and more the fact that when the Key Vault's recreated the Terraform Core bug where resources aren't picked up as requiring recreation until the second `plan` is still possible

Reply:
ah yes, either way i changed it to ID